File: PyPy3/백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
from sys import stdin
from math import isqrt

input = stdin.readline

LEN = 123_456 * 2
sieve = [False, False] + [True] * (LEN - 1)
for i in range(2, isqrt(LEN) + 1):
    if sieve[i]:
        for j in range(i * i, LEN + 1, i):
            sieve[j] = False  # 합성수 제거

while True:
    n = int(input().rstrip())
    if n == 0:
        break

    cnt = 0
    for i in range(n + 1, 2 * n + 1):
        if sieve[i]:
            cnt += 1

    print(cnt)

File: Java/백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.*;
import java.util.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static int N, cnt;
    static final int MAX = 250_000;
    static boolean[] sieve = new boolean[MAX + 1];

    public static void main(String[] args) throws IOException {

        setUp();

        while (true) {
            N = Integer.parseInt(br.readLine());
            if (N == 0)
                break;

            cnt = 0;
            for (int i = N + 1; i <= 2 * N; i++) {
                if (!sieve[i])
                    cnt++;
            }

            sb.append(cnt).append("\n");
        }

        System.out.println(sb.toString());
    }

    // 에라토스테네스의 체 구현
    static void setUp() {

        sieve[0] = true;
        sieve[1] = true;
        for (int i = 2; i <= Math.sqrt(MAX); i++) {
            if (!sieve[i]) {
                for (int j = i * i; j <= MAX; j += i) {
                    sieve[j] = true; // 합성수 처리
                }
            }
        }

    }
}
